code/modules/atmospherics/machinery/components/turbine/turbine.dm

```diff
@@ -0,0 +1,372 @@
+/obj/machinery/atmospherics/components/unary/turbine
+	icon = 'icons/obj/atmospherics/components/turbine.dmi'
+	layer = OBJ_LAYER
+	density = TRUE
+	pipe_flags = PIPING_ONE_PER_TURF | PIPING_DEFAULT_LAYER_ONLY
+	///Check to see if we are connected to the other parts
+	var/connected = FALSE
+	///volume size of our internal component
+	var/initial_volume = 500
+	///base icon for appearance changes
+	var/base_icon = ""
+
+/obj/machinery/atmospherics/components/unary/turbine/Initialize()
+	. = ..()
+	airs[1].volume = initial_volume
+
+/obj/machinery/atmospherics/components/unary/turbine/attackby(obj/item/I, mob/user, params)
+	if(!connected)
+		if(default_deconstruction_screwdriver(user, "[base_icon]", "[base_icon]", I))
```

Comment:
why not use && here? line too long?

Reply:
line would be long yes, but it's mostly for consistency between atmos components that uses this format, with some variation, because some have more checks in the first if, others in the second, so i like to have the two separated